front/src/app/core/login/guard/auth.guard.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router } from '@angular/router';
+import { Auth } from '@angular/fire/auth';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+
+  constructor(private auth: Auth, private router: Router) {}
+
+  canActivate(): boolean {
+    if (this.auth.currentUser) {
+      return true;
+    }
+    this.router.navigate(['/login']);
+    return false;
+  }
+}
```

Comment:
suggestion: depuis la v15 on peut passer par une syntaxe plus simple https://angular.dev/guide/routing/route-guards#canactivate

```suggestion
export const authGuard: CanActivateFn = () => {
  if (isDefined(inject(Auth).currentUser)) {
    return true;
  }
  return new RedirectCommand(inject(Router).parseUrl("/login"));
};
```

voir ici pour le `isDefined()` => https://github.com/sfeir-open-source/speaker-space/pull/6#discussion_r2309781443